### PR TITLE
[GOBBLIN-314] Validate filesize when copying in writer

### PR DESCRIPTION
**Pull request by jack-moseley**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-314

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):

When copying check that the number of bytes copied matches the file size 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Unit test passes

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java [hunk 1]
```diff
@@ -231,7 +231,13 @@ public boolean apply(FileStatus input) {
         if (isInstrumentationEnabled()) {
           copier.withCopySpeedMeter(this.copySpeedMeter);
         }
-        this.bytesWritten.addAndGet(copier.copy());
+        long numBytes = copier.copy();
+        long fileSize = copyableFile.getFileStatus().getLen();
```
Comment (eogren) [inline gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java]: I think this check will break if encryption is turned on in the OutputStream -- can you make sure there is a test verifying this?

Comment (jack-moseley) [reply gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java]: You're right, changed the check configurable and defaulting to false.

**Comment by jack-moseley** [comment]
@htran1 can you review?

**Comment by ibuenros** [comment]
+1

**Pull request closed without merging**